pkg/yurtmanager/webhook/yurtappset/v1beta1/yurtappset_validation.go

```diff
@@ -78,6 +106,75 @@ func (webhook *YurtAppSetHandler) ValidateUpdate(ctx context.Context, oldObj, ne
 	return nil
 }
 
+// TODO: move functions under k8s.io/kubernetes to pkg/util/kubernetes
+func (webhook *YurtAppSetHandler) validateDeployment(yas *v1beta1.YurtAppSet) error {
+	if len(yas.Spec.Workload.WorkloadTweaks) == 0 {
+		deploy := &appsv1.Deployment{}
+		deploy.Spec = *yas.Spec.Workload.WorkloadTemplate.DeploymentTemplate.Spec.DeepCopy()
+		webhook.Scheme.Default(deploy)
+		out := &apps.Deployment{}
+		if err := v1.Convert_v1_Deployment_To_apps_Deployment(deploy, out, nil); err != nil {
+			return err
+		}
+		allErrs := appsvalidation.ValidateDeploymentSpec(&out.Spec, field.NewPath("spec"), validation.PodValidationOptions{})
+		if len(allErrs) != 0 {
+			return allErrs.ToAggregate()
+		}
+		return nil
+	}
+	for _, yasTweak := range yas.Spec.Workload.WorkloadTweaks {
+		deploy := &appsv1.Deployment{}
+		deploy.Spec = *yas.Spec.Workload.WorkloadTemplate.DeploymentTemplate.Spec.DeepCopy()
+		if err := workloadmanager.ApplyTweaksToDeployment(deploy, []*v1beta1.Tweaks{&yasTweak.Tweaks}); err != nil {
+			return err
+		}
+		webhook.Scheme.Default(deploy)
+		out := &apps.Deployment{}
+		if err := v1.Convert_v1_Deployment_To_apps_Deployment(deploy, out, nil); err != nil {
+			return err
+		}
+		allErrs := appsvalidation.ValidateDeploymentSpec(&out.Spec, field.NewPath("spec"), validation.PodValidationOptions{})
+		if len(allErrs) != 0 {
+			return allErrs.ToAggregate()
+		}
+	}
```

Comment:
this code snippet can be optimized as following:

```
deploy := &appsv1.Deployment{}
deploy.Spec = *yas.Spec.Workload.WorkloadTemplate.DeploymentTemplate.Spec.DeepCopy()
for _, yasTweak := range yas.Spec.Workload.WorkloadTweaks {
    if err := workloadmanager.ApplyTweaksToDeployment(deploy, []*v1beta1.Tweaks{&yasTweak.Tweaks}); err != nil {
	return err
    }
}

webhook.Scheme.Default(deploy)
out := &apps.Deployment{}
if err := v1.Convert_v1_Deployment_To_apps_Deployment(deploy, out, nil); err != nil {
	return err
}

allErrs := appsvalidation.ValidateDeploymentSpec(&out.Spec, field.NewPath("spec"), validation.PodValidationOptions{})
if len(allErrs) != 0 {
	return allErrs.ToAggregate()
}
```

Reply:
I think we have to validate tweaks one by one. Because one WorkloadTweaks might generate many deployment yaml. For example, if one tweak's replica is -1, and next tweak's replica is normal, in this way it will pass validation when applying these tweaks directly.